src/pudl/metadata/codes.py

```diff
@@ -4,234 +4,248 @@
-            ('WND', 'Wind'),
-            ('WOC', 'Waste Coal'),
-            ('WOO', 'Waste Oil'),
-            ('WWW', 'Wood and Wood Waste'),
-        ],
-    ).convert_dtypes(),
-    "code_fixes": {},
-    "ignored_codes": [],
+    "sector_consolidated_eia": {
+        "df": pd.DataFrame(
+            columns=["code", "label", "description"],
+            data=[
+                (1, "electric_utility", "Traditional regulated electric utilities."),
+                (2, "ipp_non_cogen", "Independent power producers which are not cogenerators."),
+                (3, "ipp_cogen", "Independent power producers which are cogenerators, but whose primary business purpose is the same of electricity to the public."),
+                (4, "commercial_non_cogen", "Commercial non-cogeneration facilities that produce electric power, are connected to the grid, and can sell power to the public."),
+                (5, "commercial_cogen", "Commercial cogeneration facilities that produce electric power, are connected to the grid, and can sell power to the public."),
+                (6, "industrial_non_cogen", "Industrial non-cogeneration facilities that produce electric power, are connected to the grid, and can sell power to the public."),
+                (7, "industrial_cogen", "Industrial cogeneration facilities that produce electric power, are connected to the grid, and can sell power to the public"),
+            ]
+        ).convert_dtypes(),
+        "code_fixes": {},
+        "ignored_codes": []
+    }
 }
-"""
-Descriptive labels for aggregated fuel types used in the Annual Energy Review.
 
-See the EIA 923 Fuel Code table (Table 5) for additional information.
 """
-
-CONTRACT_TYPES_EIA: Dict[str, Any] = {
-    "df": pd.DataFrame(
-        columns=["code", "label", "description"],
-        data=[
-            ('C', 'contract', 'Fuel received under a purchase order or contract with a term of one year or longer.  Contracts with a shorter term are considered spot purchases '),  # nopep8
-            ('NC', 'new_contract', 'Fuel received under a purchase order or contract with duration of one year or longer, under which deliveries were first made during the reporting month'),  # nopep8
-            ('S', 'spot_purchase', 'Fuel obtained through a spot market purchase'),
-            ('T', 'tolling_agreement', 'Fuel received under a tolling agreement (bartering arrangement of fuel for generation)'),  # nopep8
-        ]
-    ).convert_dtypes(),
-    "code_fixes": {"N": "NC"},
-    "ignored_codes": [],
-}
+Additional descriptions on where these codes are found in documentation.
+These descriptions can't be an additional field in the entries of CODE_METADATA because those dictionaries are
+used as encoders for their respective fields.
 """
-Descriptive labels for fuel supply contract type codes reported in EIA 923.
 
-The purchase type under which fuel receipts occurred in the reporting month.
-"""
+CODE_DESCRIPTIONS = {
```

Comment:
It seems like descriptions in `RESOURCE_METADATA` are pretty scarce, for example `eia_energy_sources` has no description in `RESOURCE_METADATA` but had a pretty lengthy one in the codes metadata. One option is to put in some logic so we check for a description in `RESOURCE_METADATA` first but maybe this should just be standardized in some way.

I've changed the structure here so descriptions are part of the `CODES_METADATA` dictionary and are a new field in the Encoder class.

Reply:
They're scarce but they shouldn't be -- every single table needs to have a description in the metadata for inclusion in our documentation and output to the various other platforms where we're publishing data (like Datasette, the Intake catalogs, etc). `RESOURCE_METADATA` is the place the descriptions should go, since every Encoder also corresponds to a database table. We just haven't gotten around to filling them all in. Actually I thought I just recently merged in a PR closing #1333 but... no. Sorry, these changes are on my `eia861-metadata` branch.

Reply:
Let me run the Tox tests and see if everything is working and see if we can get it merged into `dev` to avoid this collision.